### PR TITLE
e2e: add test case for hpa replicas syncer

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
Add e2e to test hpa replicas syncer

**Which issue(s) this PR fixes**:
Fixes #none

**Special notes for your reviewer**:
none

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by chaunceyjiang — COMMENTED** [review]
Do we have the `hpaReplicasSyncer` controller enabled?

https://github.com/karmada-io/karmada/blob/master/cmd/controller-manager/app/controllermanager.go#L190

**Comment by jwcesign** [comment]
@chaunceyjiang 
It's enabled, https://github.com/karmada-io/karmada/blob/3eba2a950f9f9c6e874727c58baab6437cd3dfbd/artifacts/deploy/karmada-controller-manager.yaml#L33

File: test/e2e/hpa_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		var retainRIC *configv1alpha1.ResourceInterpreterCustomization
+
+		ginkgo.BeforeEach(func() {
+			policyNamespace = testNamespace
+			namespace = testNamespace
+			policyName = deploymentNamePrefix + rand.String(RandomStrLength)
+			deploymentName = policyName
+			hpaName = policyName
+
+			retainRIC = helper.NewRetainResourceInterpreterCustomization(namespace, policyName)
```
Comment (chaunceyjiang) [inline test/e2e/hpa_test.go:37]: I'm so sorry,  i don't understand why we need this `retainRIC` ? according to our proposal, we don't need this object. 🤔

Comment (jwcesign) [reply test/e2e/hpa_test.go]: You are right, we don't need it.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Other parts LGTM

File: test/e2e/hpa_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package e2e
```
[the remark this answers is not in the document]
Comment (jwcesign) [reply test/e2e/hpa_test.go]: fixed

**Pull request merged**